Sources/PackageModel/Manifest/Manifest.swift

```diff
@@ -560,3 +563,171 @@ extension Manifest: Encodable {
         try container.encode(self.packageKind, forKey: .packageKind)
     }
 }
+
+/// Helper methods that enable data collection through traits configurations in manifests.
+extension Manifest {
```

Comment:
suggestion (possibly-blocking): In order to ensure this extension behaves as intended, can we write `small` tests that verify the behaviour of each  [public] method in isolation, taking into account fault injections (whenever possible), boundary conditions, collections (empty, single item, two items, multiple items), etc...

If we are unable to control the test the way we want, we may need to update the production code to make it support testability.